## BEAGLE

This directory contains code to infer and phase strain haplotypes of mixed *Plasmodium falciparum* samples using [Browning and Browning (2007)](#beagle)'s method.

`BEAGLE` requires genotype likelihood of homozygous reference, heterozygous and homozygous alternative as input, which is calculated by $r \log_{10}(1-x) + a \log_{10}(x)$, where $r$ and $a$ are the reference and alternative allele counts, and $x$ is the error rate when

$x = \begin{cases}0.01, & \textrm{homozygous reference}\\ 0.5, & \textrm{heterozygous} \\ 0.99, & \textrm{homozygous alternative} \end{cases}$

First convert the VCF to `BEAGLE` input format

```bash
./convert_pf3k_vcf_to_beagle_input.py labMixed.vcf.gz
```

The we run `BEAGLE` with following command:

```bash
java  -Xmx3512m -jar ~/researchDiary/beagle/beagle.r1398.jar gl=labMixed.gl.vcf.gz out=labMixed.gl.out
```

### References

Browning, S. R. and B. L. Browning (2007) Rapid and accurate haplotype phasing and missing-data inference for whole-genome association studies by use of localised haplotype clustering. *American Journal of Human Genetics* 81(5), 1084â€“1097. <a name="beagle"></a>

